examples/nextjs-spa/pages/index.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react"
+import Link from "next/link"
+import "@ory/elements/style.css"
+
+// React
+import { useEffect, useState } from "react"
+import { useRouter } from "next/router"
+
+// Ory SDK
+import { ory } from "../components/sdk"
+
+// import css
+import styles from "../styles/Dashboard.module.css"
+import "@ory/elements/style.css"
+
+import { AxiosError } from "axios"
+import type { NextPage } from "next"
+import { createLogoutHandler } from "../pkg/hooks"
+
+const Home: NextPage = () => {
+  const [session, setSession] = useState<string>(
+    "No valid Ory Session was found.\nPlease sign in to receive one.",
+  )
+  const [hasSession, setHasSession] = useState<boolean>(false)
+  const router = useRouter()
+  const onLogout = createLogoutHandler()
+
+  useEffect(() => {
+    ory
+      .toSession()
+      .then(({ data }) => {
+        setSession(session)
+        setHasSession(true)
+      })
+      .catch((err: AxiosError) => {
+        switch (err.response?.status) {
+          case 403:
+          // This is a legacy error code thrown. See code 422 for
+          // more details.
+          case 422:
+            // This status code is returned when we are trying to
+            // validate a session which has not yet completed
+            // it's second factor
+            return router.push("/login?aal=aal2")
+          case 401:
+            // do nothing, the user is not logged in
```

Comment:
We should assume index is the "dashboard" we would like to protect and which should redirect to the login screen.